### PR TITLE
Add multi-target scrape support

### LINKED ISSUE #130: Support multi-target exporter pattern
In order to support multiple / remote memcached instances (ex, SaaS managed), it would be useful to support the [multi-target exporter pattern](https://prometheus.io/docs/guides/multi-target-exporter/).

### PR DESCRIPTION
**Pull request by SuperQ**
Add a multi-target exporter scrape endpoint.

Fixes #130
Supersedes #143

### CODE REVIEW

**Review by matthiasr — APPROVED** [review]
Closes #143

File: scraper/scraper_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	"time"
+
+	"github.com/go-kit/log"
+)
+
+func TestHandler(t *testing.T) {
+	t.Run("Success", func(t *testing.T) {
+		t.Parallel()
+
+		s := New(1*time.Second, log.NewNopLogger(), &tls.Config{})
```
Comment (matthiasr) [inline scraper/scraper_test.go:31]:
```suggestion
		s := New(1*time.Second, log.NewNopLogger(), nil)
```

We want no TLS in test, not default TLS

File: scraper/scraper_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+		if body := rr.Body.String(); !strings.Contains(body, memcachedUpMetric) {
+			t.Errorf("handler could not inspect metrics. body: %s", body)
+		}
+	})
+
+	t.Run("No target", func(t *testing.T) {
+		t.Parallel()
+
+		s := New(1*time.Second, log.NewNopLogger(), &tls.Config{})
```
Comment (matthiasr) [inline scraper/scraper_test.go:59]:
```suggestion
		s := New(1*time.Second, log.NewNopLogger(), nil)
```

File: scraper/scraper_test.go (hunk 1 above)
Comment (matthiasr) [reply scraper/scraper_test.go:31]: (I tested this locally)

Comment (SuperQ) [reply scraper/scraper_test.go]: Yea, I was just figuring that out as well. :)

**Review by matthiasr — APPROVED** [review]

**Comment by matthiasr** [comment]
Wheeeee!

**Comment by matthiasr** [comment]
Fixes #130

**Comment by chewrocca** [comment]
Thanks, this is great and works fantastic.

**Pull request merged**